cpp/ycm/CharacterRepository.h

```diff
@@ -0,0 +1,69 @@
+// Copyright (C) 2018 ycmd contributors
+//
+// This file is part of ycmd.
+//
+// ycmd is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// ycmd is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with ycmd.  If not, see <http://www.gnu.org/licenses/>.
+
+#ifndef CHARACTER_REPOSITORY_H_36TXTS6C
+#define CHARACTER_REPOSITORY_H_36TXTS6C
+
+#include "Character.h"
+
+#include <memory>
+#include <mutex>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace YouCompleteMe {
+
+using CharacterHolder = std::unordered_map< std::string,
+                                            std::unique_ptr< Character > >;
```

Comment:
Couldn't the value be stored as an object? I'm not sure unique_ptr is needed here